tests/test_query.py

```diff
@@ -565,16 +614,19 @@ def test_get_opposite_dub_action(input, output):
         user_generator,
         playback_generator,
         user_action_generator,
-        loops,
-        output
-):
-    actions = ['upvote', 'downvote', 'upvote', 'skip']
+        user_action,
+        expected_result):
     track = await track_generator()
-    user = await user_generator()
-    playback = await playback_generator(user=user, track=track)
-    for _, action in zip(range(loops), actions):
-        await user_action_generator(user=user, playback=playback, action=action)
-
-    user_actions = await query_simplified_user_actions(playback_id=playback['id'], conn=db_conn)
-    result = {ua['action'] for ua in user_actions}
-    assert result == output
+    registered_user = await user_generator()
+    playback = await playback_generator(user=registered_user, track=track)
+
+    for action in user_action:
+        if action['user'] is None:
+            user = await user_generator()
```

Comment:
Just create this statically, like 3 users and that's it

Reply:
If we create the users statically, we still need to somehow map the `action['user']` value to the corresponding user (`user1`, `user2`...). Is it not too verbose?
```
user1 = await user_generator()
user2 = await user_generator()
user3 = await user_generator()
user4 = await user_generator()

if action['user'] == 1:
    user = user1
elif action['user'] == 2:
    user = user2
elif action['user'] == 3:
    user = uset3
elif action['user'] == 4:
    user = user4
```

Reply:
Check that user_action_generator just needs a dict with an 'id' key, therefore you just need to call it like
```
await user_action_generator(user={'id': action['user_id']}, ...)
```

Reply:
I was actually looking for that, but I got confused with the function signature (`def user_action_generator(db_conn):`). I missed that all the arguments extra arguments are passed down to the `generate_user_action` function via `*`... #$@&%*!